app/screens/terms_of_service/terms_of_service.js

```diff
@@ -0,0 +1,272 @@
+
+        this.setState({
+            termsId: '',
+            termsText: '',
+            loading: true,
+        });
+
+        const {data} = await actions.getTermsOfService();
+        if (data) {
+            this.setState({
+                termsId: data.id,
+                termsText: data.text,
+                loading: false,
+            }, () => {
+                this.setNavigatorButtons(true);
+            });
+        } else {
+            this.setState({
+                loading: false,
+            }, () => {
+                this.setNavigatorButtons(true);
+            });
+            Alert.alert(
+                this.props.siteName,
+                intl.formatMessage({
+                    id: 'mobile.terms_of_service.get_terms_error',
+                    defaultMessage: 'Unable to load terms of service. If this issue persists, contact your System Administrator.',
+                }),
+                [{
+                    text: intl.formatMessage({id: 'mobile.terms_of_service.alert_ok', defaultMessage: 'Ok'}),
+                    onPress: async () => {
+                        await actions.logout();
+                        this.setNavigatorButtons(true);
+                        this.props.navigator.dismissAllModals();
+                    },
+                }],
+            );
+        }
+    };
+
+    handleAcceptTerms = async () => {
+        await this.registerUserAction(
+            true,
+            () => {
+                this.props.navigator.dismissModal({
+                    animationType: 'slide-down',
+                });
+            }
+        );
+    };
+
+    handleRejectTerms = async () => {
+        const {actions, siteName} = this.props;
+        const {intl} = this.context;
+
+        await this.registerUserAction(
+            false,
+            () => {
+                Alert.alert(
+                    this.props.siteName,
+                    intl.formatMessage({
+                        id: 'mobile.terms_of_service.terms_rejected',
+                        defaultMessage: 'You must agree to the terms of service before accessing {siteName}. Please contact your System Administrator for more details.',
+                    }, {
+                        siteName,
+                    }),
+                    [{
+                        text: intl.formatMessage({id: 'mobile.terms_of_service.alert_ok', defaultMessage: 'Ok'}),
+                        onPress: async () => {
+                            await actions.logout();
+                            this.setNavigatorButtons(true);
+                            this.props.navigator.dismissAllModals();
+                        },
+                    }],
+                );
+            }
+        );
+    };
+
+    registerUserAction = async (accepted, success) => {
+        const {actions} = this.props;
+        const {intl} = this.context;
+
+        this.setNavigatorButtons(false);
+        this.setState({
+            loading: true,
```

Comment:
So whenever I press either button the text dissapears and turns into an activity indicator, is there a better way to handle this? To me it feels really strange, perhaps we can discuss this with the UX team?

Reply:
I actually thought this was a good idea. Since we cannot change the navigator buttons to spinners and even if we could, it would not look good on mobile. And showing a spinner while some action is being performed might be good UX.
We also do something similar in `create channel modal`.
We could, of course, change this if we have something better.

@jasonblais Thoughts?

Reply:
@jasonblais The current behaviour is that when the user clicks I Agree, then the entire text disappears and is replaced by a loading spinner while the API call is being made.

What @enahum probably meant was that it is strange UX.

But this behaviour is similar to `create channel modal`.

So, should we keep the current behaviour or change it after discussion with the UX team?

Reply:
I see, I misread the conversation. I would personally like to test it out and share with the UX team.